lms/static/js/student_profile/views/learner_profile_factory.js

```diff
@@ -121,6 +125,15 @@
                 })
             ];
 
+            var badgeCollection = new PagingCollection();
+            badgeCollection.url = options.badges_api_url || "/dummy/url";
```

Comment:
@nasthagiri Yes. The collection won't initialize without a URL, but if we don't have one, we won't ever use it, because it means badges are disabled.